test/integration/PartialClosures.t.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.22;
+
+import { console2 as console } from "forge-std/console2.sol";
+import { FixedPointMath } from "hyperdrive/contracts/src/libraries/FixedPointMath.sol";
+import { Lib } from "hyperdrive/test/utils/Lib.sol";
+import { HyperdriveUtils } from "hyperdrive/test/utils/HyperdriveUtils.sol";
+import { EverlongTest } from "../harnesses/EverlongTest.sol";
+import { IEverlong } from "../../contracts/interfaces/IEverlong.sol";
+import { HyperdriveExecutionLibrary } from "../../contracts/libraries/HyperdriveExecution.sol";
+
+contract PartialClosures is EverlongTest {
+    using FixedPointMath for uint256;
+    using Lib for *;
+    using HyperdriveUtils for *;
+    using HyperdriveExecutionLibrary for *;
+
+    /// @dev Tests that immature positions can be partially closed when
+    ///      additional liquidity is needed to service a withdrawal.
+    ///      Ensures that the amount of longs closed in the position is not
+    ///      excessive.
+    function testFuzz_partial_closures(
+        uint256 _targetIdle,
+        uint256 _maxIdle,
+        uint256 _deposit,
+        uint256 _redemptionPercentage
+    ) external {
+        TARGET_IDLE_LIQUIDITY_PERCENTAGE = bound(_targetIdle, 0.001e18, 0.5e18);
+        MAX_IDLE_LIQUIDITY_PERCENTAGE = bound(
+            _maxIdle,
+            TARGET_IDLE_LIQUIDITY_PERCENTAGE + 0.001e18,
+            0.9e18
+        );
+        deployEverlong();
+
+        // Alice deposits into Everlong.
+        uint256 aliceDepositAmount = bound(
+            _deposit,
+            MINIMUM_TRANSACTION_AMOUNT * 1000,
```

Comment:
I'd leave a TODO here to investigate why this is so much higher than the equivalent bounds in our Hyperdrive tests

Reply:
This is actually just due to the structure of the test....

So for partial closures, the remaining bonds after the partial close have to be worth at least `minTxAmount`. 

- If we allow alice to deposit `minTxAmount`, then there's no way her redemption could trigger a partial closure.
- If we allow alice to deposit `2*minTxAmount`, then she can only redeem a maximum of ~50% of her position and trigger a partial closure.
- `5*minTxAmount` means max redeem of 80%
and so on....

I just chose 1000 here and a max redemption of 90% to be safe.... that being said, doing something like 100 and 89% would enable more fringe values to pop up, so I'll update it accordingly :)